src/store/consts/utils.ts

```diff
@@ -235,27 +244,37 @@ export const createPlaceholderLiquidityPlot = (
   return isXtoY ? ticksData : ticksData.reverse()
 }
 
-export interface CoingeckoPriceData {
-  price: number
-  priceChange: number
-}
-export interface CoingeckoApiPriceData {
+export type CoinGeckoAPIData = CoinGeckoAPIPriceData[]
+
+export type CoinGeckoAPIPriceData = {
   id: string
   current_price: number
   price_change_percentage_24h: number
 }
 
-export const getCoingeckoTokenPrice = async (id: string): Promise<CoingeckoPriceData> => {
-  return await axios
-    .get<
-      CoingeckoApiPriceData[]
-    >(`https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&ids=${id}`)
-    .then(res => {
-      return {
-        price: res.data[0].current_price ?? 0,
-        priceChange: res.data[0].price_change_percentage_24h ?? 0
-      }
-    })
+const COINGECKO_QUERY_COOLDOWN = 20 * 60 * 1000
+
+export const getCoingeckoTokenPrice = async (id: string): Promise<number | undefined> => {
```

Comment:
I think it's worth considering keeping the prices in the store

Reply:
we can do that but it would require us to have the same state in two places (local storage and store) at the same time and take care of all updates to make sure it's synced + it would require more changes to the code (now the usage is almost the same so replacing one line in each function execution was easy)

Reply:
the Store will be the only place where the application will have access to prices. The attempt to refresh the price would have access with local storage

But the argument about more effort makes sense, so let's leave it as it is with access to local storage without store